llvm/lib/Transforms/Vectorize/SLPVectorizer.cpp

```diff
@@ -2366,7 +2363,10 @@ class BoUpSLP {
     }
 
     /// \returns the number of operands.
-    unsigned getNumOperands() const { return OpsVec.size(); }
+    /// reorder can only work on instructions with 2 operands. If we use
+    /// OpsVec.size() here, reorder will swap non-swappable operands (because
+    /// APO is a boolean value).
+    unsigned getNumOperands() const { return 2; }
```

Comment:
I don't like this approach. I plan to extend the reordering to support many operands, better to avoid constant value here.

Reply:
Then we still need arg_size here.
On the other hand, I am curious about how reorder support more operands. Only the first two operands of BinaryOperator, CmpInst and IntrinsicInst are commutative. What kinds of instructions can be supported if we enhance reorder?

Reply:
if you have `add i32 (add p1, p2), (add p3, p4)`, we can model it as `add p1, p2, p3, p4` and reorder operands.